subsys/bluetooth/host/att.c

```diff
@@ -176,30 +173,20 @@ static struct bt_att_req cancel;
  */
 static k_tid_t att_handle_rsp_thread;
 
-#define bt_att_tx_meta_data(buf) (((struct bt_att_tx_meta *)net_buf_user_data(buf))->data)
+static struct bt_att_tx_meta_data tx_meta_data_storage[CONFIG_BT_ATT_TX_COUNT];
 
-static struct bt_att_tx_meta_data tx_meta_data_storage[CONFIG_BT_CONN_TX_MAX];
-K_FIFO_DEFINE(free_att_tx_meta_data);
+NET_BUF_POOL_DEFINE(att_pool, CONFIG_BT_ATT_TX_COUNT,
+		    BT_L2CAP_SDU_BUF_SIZE(BT_ATT_BUF_SIZE),
+		    CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
```

Comment:
As discussed offline: Here we can add a destroy cb that memsets metadata to 0xff[^1] to gain confidence that the metadata is not accessed while the net buf is not allocated.

I think the cb can be guarded by CONFIG_TESTING.

[^1]: By using a value other than 0, we can trip up any null checks. Those null checks should also not happen on a unallocated metadata.

Reply:
done